### DISCUSSION TITLE
Using a time based pagination cursor in outgoing messages

### DISCUSSION THREAD

**Discussion by reshadman** [postalserver/postal#1488] [Feature suggestions]
It would be nice to implement a time based pagination cursor instead of the current `limit` `ofset` strategy for outgoing messages. As the number of outgoing messages increases the query gets slower and slower.

I don't know Rails enough to implement the functionality, I would appreciate if you guide somehow on the implementation or consider the problem.

**Comment by willpower232** [comment]
definitely worth looking at, out of curiousity, how many messages do you have?

**Comment by reshadman** [comment]
We send around 700K per day with retention days of: 60. Working like a charm with 30 workers. Nice piece of software btw saved us around $1500 per month with a single $40 server and a simple IP warm up strategy.
